src/agent/backgroundAgent.ts

```diff
@@ -18,12 +19,10 @@ process.on('message', async (startOptions: string) => {
   // Split the message into password and string
   const ops = JSON.parse(startOptions);
   try {
-    polykeyAgent = await PolykeyAgent.createPolykeyAgent({
-      password: ops.password,
-      nodePath: ops.nodePath,
-    });
+    polykeyAgent = await PolykeyAgent.createPolykeyAgent(ops);
+    recoveryCode = polykeyAgent.generatedRecoveryCode;
     // Catching kill signals.
-    process.send && process.send('started');
+    process.send && process.send(`started:${recoveryCode}`);
```

Comment:
What is this?

Reply:
We need to get the recovery code from the agent when it is created in the background.

Reply:
I don't understand, can you explain why this requires `process.send`? I thought you'd be using the `WorkerManager`.

Reply:
it's the background agent, it needs to send the information back to the CLI command to let it know that it succeeded or failed to start or generated a recovery code.

Reply:
If it's a backgrounded agent, how does the foreground command exit?

Alternatively I feel like you should get bootstrapping working in foreground first before backgrounding the agent. Meaning key setup can be done in the bootstrap foreground process.

Reply:
As per the discussion this will be addressed in a separate issue.

Reply:
Ok I just had a look over the process system. Nodejs provides an IPC channel between parent and child processes. I didn't know it had this ability. It may be using a pipe/named pipe or unix domain socket underneath to implement this as I've done something like this in C before. I need to evaluate this to see what we need to do here.